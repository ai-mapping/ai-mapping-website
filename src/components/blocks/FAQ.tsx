import React from 'react';
import FAQAccordion from './FAQAccordion';
import Section from '@/components/layout/Section';

// Define types for FAQ items
export type FAQItem = {
  question: string;
  answer: string;
};

// Main props interface
export type FAQProps = {
  heading: string;
  faqItems: FAQItem[];
  askButtonText: string;
};

// Default props
export const defaultFAQProps: FAQProps = {
  heading: 'Часто задаваемые вопросы',
  faqItems: [
    {
      question: 'Каковы сроки доставки запчастей из Китая?',
      answer:
        'Стандартный срок доставки запчастей составляет 7-14 дней. Точный срок зависит от наличия детали у наших партнеров в Китае, сложности ее поиска и логистических особенностей. В экстренных случаях возможна ускоренная доставка за дополнительную плату.',
    },
    {
      question: 'Насколько качественны аналоги запчастей?',
      answer:
        'Мы поставляем только качественные аналоги, произведенные на сертифицированных заводах в Китае. По своим техническим характеристикам и надежности они сопоставимы с оригинальными запчастями, при этом стоят значительно дешевле. На все запчасти предоставляется гарантия.',
    },
    {
      question: 'Какие способы оплаты вы принимаете?',
      answer:
        'Мы работаем как с наличным, так и с безналичным расчетом. Для юридических лиц доступна оплата по счету с НДС. Обычно мы работаем по предоплате 50%, остаток – перед отгрузкой товара. Для постоянных клиентов возможны индивидуальные условия оплаты.',
    },
    {
      question: 'Что делать, если нужна очень редкая запчасть?',
      answer:
        'Мы специализируемся на поиске редких и труднодоступных деталей. Наши партнеры в Китае имеют доступ к широкой сети производителей и складов. Даже если запчасть отсутствует в наличии, во многих случаях возможно её изготовление под заказ. Для таких случаев мы рекомендуем предоставить нам максимум информации о детали, включая фото, артикул и модель техники.',
    },
  ],
  askButtonText: 'Задать свой вопрос',
};

const FAQ = ({
  heading = defaultFAQProps.heading,
  faqItems = defaultFAQProps.faqItems,
  askButtonText = defaultFAQProps.askButtonText,
}: Partial<FAQProps>) => {
  return (
    <Section background="background" paddingY="lg" containerWidth="narrow">
      <h2 className="text-2xl md:text-3xl font-semibold text-center mb-8 text-white">
        {heading}
      </h2>

      <FAQAccordion faqItems={faqItems} askButtonText={askButtonText} />
    </Section>
  );
};

export default FAQ;
