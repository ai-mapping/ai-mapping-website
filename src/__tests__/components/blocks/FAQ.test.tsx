import React from 'react';
import { render, screen } from '@testing-library/react';
import FAQ, { defaultFAQProps } from '@/components/blocks/FAQ';

// Mock the FAQAccordion component
jest.mock('@/components/blocks/FAQAccordion', () => {
  return {
    __esModule: true,
    default: ({ faqItems, askButtonText }) => (
      <div data-testid="faq-accordion">
        <div>FAQ Items: {faqItems.length}</div>
        <div>
          {faqItems.map((item, index) => (
            <div key={index} data-testid={`faq-item-${index}`}>
              <div>Question: {item.question}</div>
              <div>Answer: {item.answer}</div>
            </div>
          ))}
        </div>
        <button data-testid="ask-button">{askButtonText}</button>
      </div>
    )
  };
});

describe('FAQ Component', () => {
  it('renders with default props correctly', () => {
    const { container } = render(<FAQ />);
    
    // Check if heading is rendered
    expect(screen.getByText(defaultFAQProps.heading)).toBeInTheDocument();
    
    // Check if the FAQAccordion is rendered
    expect(screen.getByTestId('faq-accordion')).toBeInTheDocument();
    expect(screen.getByText(`FAQ Items: ${defaultFAQProps.faqItems.length}`)).toBeInTheDocument();
    
    // Check if all FAQ items are rendered
    defaultFAQProps.faqItems.forEach((item, index) => {
      expect(screen.getByTestId(`faq-item-${index}`)).toBeInTheDocument();
      expect(screen.getByText(`Question: ${item.question}`)).toBeInTheDocument();
      expect(screen.getByText(`Answer: ${item.answer}`)).toBeInTheDocument();
    });
    
    // Check if the ask button is rendered with correct text
    expect(screen.getByTestId('ask-button')).toBeInTheDocument();
    expect(screen.getByText(defaultFAQProps.askButtonText)).toBeInTheDocument();
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
  
  it('renders with custom props correctly', () => {
    const customProps = {
      heading: 'Custom FAQ Heading',
      faqItems: [
        {
          question: 'Custom question?',
          answer: 'Custom answer.'
        }
      ],
      askButtonText: 'Custom Ask Button'
    };
    
    const { container } = render(<FAQ {...customProps} />);
    
    // Check if custom heading is rendered
    expect(screen.getByText(customProps.heading)).toBeInTheDocument();
    
    // Check if FAQ items count is correct
    expect(screen.getByText(`FAQ Items: ${customProps.faqItems.length}`)).toBeInTheDocument();
    
    // Check if the custom FAQ item is rendered
    expect(screen.getByTestId('faq-item-0')).toBeInTheDocument();
    expect(screen.getByText(`Question: ${customProps.faqItems[0].question}`)).toBeInTheDocument();
    expect(screen.getByText(`Answer: ${customProps.faqItems[0].answer}`)).toBeInTheDocument();
    
    // Check if the ask button has custom text
    expect(screen.getByText(customProps.askButtonText)).toBeInTheDocument();
    
    // Snapshot test
    expect(container).toMatchSnapshot();
  });
});