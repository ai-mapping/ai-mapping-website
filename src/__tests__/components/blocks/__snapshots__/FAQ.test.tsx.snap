// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ Component renders with custom props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 max-w-3xl"
    >
      <h2
        class="text-2xl md:text-3xl font-semibold text-center mb-8 text-white"
      >
        Custom FAQ Heading
      </h2>
      <div
        data-testid="faq-accordion"
      >
        <div>
          FAQ Items: 
          1
        </div>
        <div>
          <div
            data-testid="faq-item-0"
          >
            <div>
              Question: 
              Custom question?
            </div>
            <div>
              Answer: 
              Custom answer.
            </div>
          </div>
        </div>
        <button
          data-testid="ask-button"
        >
          Custom Ask Button
        </button>
      </div>
    </div>
  </section>
</div>
`;

exports[`FAQ Component renders with default props correctly 1`] = `
<div>
  <section
    class="bg-background py-12"
  >
    <div
      class="container mx-auto px-4 md:px-6 lg:px-8 max-w-3xl"
    >
      <h2
        class="text-2xl md:text-3xl font-semibold text-center mb-8 text-white"
      >
        Часто задаваемые вопросы
      </h2>
      <div
        data-testid="faq-accordion"
      >
        <div>
          FAQ Items: 
          4
        </div>
        <div>
          <div
            data-testid="faq-item-0"
          >
            <div>
              Question: 
              Каковы сроки доставки запчастей из Китая?
            </div>
            <div>
              Answer: 
              Стандартный срок доставки запчастей составляет 7-14 дней. Точный срок зависит от наличия детали у наших партнеров в Китае, сложности ее поиска и логистических особенностей. В экстренных случаях возможна ускоренная доставка за дополнительную плату.
            </div>
          </div>
          <div
            data-testid="faq-item-1"
          >
            <div>
              Question: 
              Насколько качественны аналоги запчастей?
            </div>
            <div>
              Answer: 
              Мы поставляем только качественные аналоги, произведенные на сертифицированных заводах в Китае. По своим техническим характеристикам и надежности они сопоставимы с оригинальными запчастями, при этом стоят значительно дешевле. На все запчасти предоставляется гарантия.
            </div>
          </div>
          <div
            data-testid="faq-item-2"
          >
            <div>
              Question: 
              Какие способы оплаты вы принимаете?
            </div>
            <div>
              Answer: 
              Мы работаем как с наличным, так и с безналичным расчетом. Для юридических лиц доступна оплата по счету с НДС. Обычно мы работаем по предоплате 50%, остаток – перед отгрузкой товара. Для постоянных клиентов возможны индивидуальные условия оплаты.
            </div>
          </div>
          <div
            data-testid="faq-item-3"
          >
            <div>
              Question: 
              Что делать, если нужна очень редкая запчасть?
            </div>
            <div>
              Answer: 
              Мы специализируемся на поиске редких и труднодоступных деталей. Наши партнеры в Китае имеют доступ к широкой сети производителей и складов. Даже если запчасть отсутствует в наличии, во многих случаях возможно её изготовление под заказ. Для таких случаев мы рекомендуем предоставить нам максимум информации о детали, включая фото, артикул и модель техники.
            </div>
          </div>
        </div>
        <button
          data-testid="ask-button"
        >
          Задать свой вопрос
        </button>
      </div>
    </div>
  </section>
</div>
`;
